feature-libs/checkout/scheduled-replenishment/components/services/checkout-replenishment-form-service.ts

```diff
@@ -25,17 +44,74 @@ export class CheckoutReplenishmentFormService {
   private scheduleReplenishmentFormData$: BehaviorSubject<ScheduleReplenishmentForm> =
     new BehaviorSubject<ScheduleReplenishmentForm>(this.defaultFormData);
 
-  constructor() {}
+  protected orderType$ = new BehaviorSubject<ORDER_TYPE>(
+    ORDER_TYPE.PLACE_ORDER
+  );
+
+  constructor(protected eventService: EventService) {
+    this.registerOrderTypeEventListers();
+  }
+
+  protected registerOrderTypeEventListers(): void {
+    this.subscriptions.add(
+      merge(
+        this.eventService.get(DeliveryAddressSetEvent),
+        this.eventService.get(LogoutEvent),
+        this.eventService.get(LoginEvent),
+        this.eventService.get(DeliveryAddressClearedEvent),
+        this.eventService.get(DeliveryModeSetEvent),
+        this.eventService.get(DeliveryModeClearedEvent),
+        this.eventService.get(SaveCartSuccessEvent),
+        this.eventService.get(RestoreSavedCartSuccessEvent),
+        this.eventService.get(PaymentDetailsCreatedEvent),
+        this.eventService.get(PaymentDetailsSetEvent),
+        this.eventService.get(OrderPlacedEvent),
+        this.eventService.get(ReplenishmentOrderScheduledEvent),
+        this.eventService.get(MergeCartSuccessEvent)
+      ).subscribe(() => {
+        this.orderType$.next(ORDER_TYPE.PLACE_ORDER);
```

Comment:
Because of the above changes, we need to cast it:
```suggestion
      (this.orderType$ as BehaviorSubject<ORDER_TYPE>).next(
          ORDER_TYPE.PLACE_ORDER
        );
```